interfaces/builtin/mir.go

```diff
@@ -0,0 +1,606 @@
+# limits
+setrlimit
+prlimit64
+
+set_mempolicy
+set_robust_list
+setsid
+set_tid_address
+setsockopt
+setxattr
+fsetxattr
+lsetxattr
+
+shmat
+shmctl
+shmdt
+shmget
+signal
+sigaction
+signalfd
+signalfd4
+sigaltstack
+sigpending
+sigprocmask
+sigreturn
+sigsuspend
+sigtimedwait
+sigwaitinfo
+socket
+socketpair
+splice
+
+stat
+stat64
+fstat
+fstat64
+fstatat64
+lstat
+newfstatat
+oldfstat
+oldlstat
+oldstat
+
+statfs
+statfs64
+fstatfs
+fstatfs64
+statvfs
+fstatvfs
+ustat
+
+symlink
+symlinkat
+
+sync
+sync_file_range
+sync_file_range2
+arm_sync_file_range
+fdatasync
+fsync
+syncfs
+sysinfo
+syslog
+tee
+tgkill
+time
+timer_create
+timer_delete
+timer_getoverrun
+timer_gettime
+timer_settime
+timerfd_create
+timerfd_gettime
+timerfd_settime
+times
+tkill
+
+truncate
+truncate64
+ftruncate
+ftruncate64
+
+umask
+
+uname
+olduname
+oldolduname
+
+unlink
+unlinkat
+
+utime
+utimensat
+utimes
+futimesat
+
+vfork
+vmsplice
+wait4
+oldwait4
+waitpid
+waitid
+
+write
+writev
+pwrite
+pwrite64
+pwritev
+`)
+
+var mirConnectedPlugAppArmor = []byte(`
+# Description: Allow use of the Mir server. Reserved because this
+# gives privileged access to the system.
+# Usage: reserved
+
+
+capability dac_override,
+capability sys_tty_config,
+capability sys_admin,
+
+unix (receive, send) type=seqpacket addr=none,
+/tmp/mir_socket rw,
+/dev/dri/card0 rw,
+/dev/shm/\#* rw,
+
+/sys/devices/**/uevent rw,
+/sys/devices/**/ r,
+/dev/input/* rw,
+/dev/tty* wr,
+/run/udev/data/* r,
+/run/udev/** rw,
+
+ptrace peer=**,
+/bin/sleep mrix,
+/bin/pidof mrix,
+/bin/sed mrix,
+/bin/cp mrix,
+/sbin/killall5 ixr,
+/usr/bin/expr ixr,
+/usr/bin/chmod ixr,
+/bin/chmod ixr,
+/proc/ r,
+/proc/*/stat r,
+/proc/*/cmdline r,
+/sys/bus/ r,
+/sys/class/ r,
+/sys/class/input/ r,
+/sys/class/drm/ r,
+/etc/udev/udev.conf r,
+capability sys_ptrace,
+capability chown,
+capability fowner,
+capability sys_ptrace,
+`)
+
+var mirConnectedPlugSecComp = []byte(`
+# Description: Allow operating as the mir service. Reserved because this
+# gives privileged access to the system.
+
+`)
+
+type MirInterface struct{}
+
+func (iface *MirInterface) Name() string {
+	return "mir"
+}
+
+func (iface *MirInterface) PermanentPlugSnippet(
+            plug *interfaces.Plug, 
+            securitySystem interfaces.SecuritySystem) ([]byte, error) {
+	switch securitySystem{
+        case interfaces.SecurityAppArmor:
+                return mirPermanentSlotAppArmor, nil
+        case interfaces.SecuritySecComp:
+                return mirPermanentSlotSecComp, nil
+	case interfaces.SecurityUDev, interfaces.SecurityDBus:
+		return nil, nil
+	default:
+		return nil, interfaces.ErrUnknownSecurity
+	}
+}
+
+func (iface *MirInterface) ConnectedPlugSnippet(
+            plug *interfaces.Plug, 
+            slot *interfaces.Slot, 
+            securitySystem interfaces.SecuritySystem) ([]byte, error) {
+	switch securitySystem {
+	case interfaces.SecurityAppArmor:
```

Comment:
correct, after lots and lots of experimenting - returning the seccomp and aa profiles for mir server on PermanentPlugSnippet is what worked, returning on any other (PermSlotSnippet, ConnectedSlotSnippet, ConnectedPlugSnippet did not work and i tested  each). Which honestly was not what i thought would occur - I believed that for Mir _server_ it should have reported on PermanentSlot